src/features/comments/comments.ts

```diff
@@ -0,0 +1,91 @@
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Comment } from '../../types/Comment';
+import {
+  createComment,
+  deleteComment,
+  getPostComments,
+} from '../../api/comments';
+
+type InitialState = {
+  comments: Comment[];
+  loaded: boolean;
+  hasError: boolean;
+};
+
+const initialState: InitialState = {
+  comments: [],
+  loaded: false,
+  hasError: false,
+};
+
+export const init = createAsyncThunk('comments/fetch', (postId: number) => {
+  return getPostComments(postId);
+});
+
+export const deleteComments = createAsyncThunk(
+  'comments/delete',
+  async (commentId: number) => {
+    await deleteComment(commentId);
+
+    return commentId;
+  },
+);
+
+export const addComments = createAsyncThunk(
+  'comments/post',
+  (comment: Comment) => {
+    const newComment = createComment(comment);
+
+    return newComment;
```

Comment:
The `addComments` thunk should await the `createComment` function to ensure the comment is created before returning it. Consider changing `const newComment = createComment(comment);` to `const newComment = await createComment(comment);`.